src/content/tutorial/ui/5-user-input.md

```diff
@@ -0,0 +1,646 @@
+correctly.  Submitting the guess requires us to use the functionality
+of a `StatefulWidget`, which you’ll do in the [`StatefulWidget` lesson][].
+
+## Buttons
+
+To improve the UX on mobile and reflect well-known UI, there should
+also be a button that can submit the guess.
+
+{%- comment %}
+TODO(ewindmill) embed video
+{%- endcomment %}
+
+There are many button widgets built into Flutter, like [`TextButton`][],
+[`ElevatedButton`][], and the button you’ll use now: [`IconButton`][].  All of
+these buttons (and many other interaction widgets) require two
+arguments (in addition to their optional arguments): 
+
+* A callback function passed to `onPressed`.   
+* A widget that makes up the content of the button (often a `Text` or `Icon`).
+
+Add an icon button to the row widget’s children list in the
+`GuessInput` widget, and give it an [`Icon`][] widget to display. The
+`Icon` widget itself requires icon configuration, and in this case the
+`padding` property, which sets the padding betweenthe edge of the
+button and the icon it wraps, is set to zero. This removes the default
+padding and makes the button smaller.
+
+```dart
+class GuessInput extends StatelessWidget {
+  GuessInput({super.key, required this.onSubmitGuess});
+
+  final void Function(String) onSubmitGuess;
+
+  final TextEditingController _textEditingController = TextEditingController();
+  final FocusNode _focusNode = FocusNode();
+
+    @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(...),
+        IconButton(
+          padding: EdgeInsets.zero,
+          icon: Icon(Icons.arrow_circle_up),  
+        ),
+      ],
+    );
+  }
+}
+```
+
+The `IconButton.onPressed` callback should look familiar:
+
+```dart
+class GuessInput extends StatelessWidget {
+  GuessInput({super.key, required this.onSubmitGuess});
+
+  final void Function(String) onSubmitGuess;
+
+  final TextEditingController _textEditingController = TextEditingController();
+  final FocusNode _focusNode = FocusNode();
+
+    @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(...),
+        IconButton(
+          padding: EdgeInsets.zero,
+          icon: Icon(Icons.arrow_circle_up),
+          onPressed: () {
+            onSubmitGuess(_textEditingController.text.trim());
+            _textEditingController.clear();
+            _focusNode.requestFocus();
+          },
+        ),
+      ],
+    );
+  }
+}
+```
+
+This method is doing the same as the `onSubmitted` callback on the `TextField`. 
+
+:::note Challenge - Share "on submitted" logic.
+
+You may be thinking, "Shouldn’t we abstract these methods into one
+function and pass it to both inputs?" You could, but it wouldn’t be as
+clean as you might think because the callbacks to
+`IconButton.onPressed` and `TextField.onSubmitted` require different
+arguments. That said, if the logic was more complex, it'd be better to
+do so. 
```

Comment:
```suggestion
arguments. That said, if the logic was more complex, it would be better to
do so. 
```

Really? Why?

Reply:
Resolved by making the whole thing more concise in a separate commit